File: spark-study/src/main/java/com/cfy/utils/ConfigurationManager.java
package com.cfy.utils;

import java.io.IOException;
import java.io.InputStream;
import java.util.Properties;

/**
 * Created by IntelliJ IDEA.
 * User: Leo_Chan
 * Date: 2018/6/30
 * Time: 22:06
 * Work contact: Astion_Leo@163.com
 */


public class ConfigurationManager {


    private static Properties properties = new Properties();

    static {
        try {
            InputStream inputStream = ConfigurationManager.class.getClassLoader().getResourceAsStream("my.properties");
            properties.load(inputStream);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }


    public static String getString(String key) {
        String value = properties.getProperty(key);
        return value;
    }


    public static Integer getInteger(String key) {

        try {
            String value = properties.getProperty(key);
            return Integer.valueOf(value);
        } catch (NumberFormatException e) {
            e.printStackTrace();
        }
        return 0;
    }

    public static Boolean getBoolean(String key) {
        try {
            String value = properties.getProperty(key);
            return Boolean.valueOf(value);
        } catch (Exception e) {
            e.printStackTrace();
        }
        return false;
    }
}

File: kafka2hive-new/src/main/scala/com/cfy/config/ConfigurationManager.scala
package com.cfy.config

import java.io.FileInputStream
import java.util.Properties

object ConfigurationManager {

  def load(path: String) = {
    val prop = new Properties()
    prop.load(new FileInputStream(path))
    prop
  }

//  def getString(key: String) = { Option(prop.getProperty(key)) }
//
//  def getInt(key: String) = { Option(prop.get(key).toString.toInt) }


}
